viper/parser/stmt.py

```diff
@@ -0,0 +1,269 @@
+            if not target.mutable:
+                raise ConstancyViolationException("Cannot modify function argument", self.stmt.targets[0])
+            o = make_setter(target, sub, target.location)
+        o.pos = getpos(self.stmt)
+        return o
+
+    def parse_if(self):
+        from .parser import (
+            parse_value_expr,
+            parse_body,
+        )
+        if self.stmt.orelse:
+            add_on = [parse_body(self.stmt.orelse, self.context)]
+        else:
+            add_on = []
+        return LLLnode.from_list(['if', parse_value_expr(self.stmt.test, self.context), parse_body(self.stmt.body, self.context)] + add_on, typ=None, pos=getpos(self.stmt))
+
+    def call(self):
+        from .parser import (
+            parse_expr,
+            pack_arguments,
+            pack_logging_data,
+            pack_logging_topics,
+            external_contract_call_stmt,
+        )
+        if isinstance(self.stmt.func, ast.Name) and self.stmt.func.id in stmt_dispatch_table:
+                return stmt_dispatch_table[self.stmt.func.id](self.stmt, self.context)
+        elif isinstance(self.stmt.func, ast.Attribute) and isinstance(self.stmt.func.value, ast.Name) and self.stmt.func.value.id == "self":
+            if self.stmt.func.attr not in self.context.sigs['self']:
+                raise VariableDeclarationException("Function not declared yet (reminder: functions cannot "
+                                                    "call functions later in code than themselves): %s" % self.stmt.func.attr)
+            inargs, inargsize = pack_arguments(self.context.sigs['self'][self.stmt.func.attr],
+                                                [parse_expr(arg, self.context) for arg in self.stmt.args],
+                                                self.context)
+            return LLLnode.from_list(['assert', ['call', ['gas'], ['address'], 0, inargs, inargsize, 0, 0]],
+                                        typ=None, pos=getpos(self.stmt))
+        elif isinstance(self.stmt.func, ast.Attribute) and isinstance(self.stmt.func.value, ast.Call):
+            return external_contract_call_stmt(self.stmt, self.context)
+        elif isinstance(self.stmt.func, ast.Attribute) and self.stmt.func.value.id == 'log':
+            if self.stmt.func.attr not in self.context.sigs['self']:
+                raise VariableDeclarationException("Event not declared yet: %s" % self.stmt.func.attr)
+            event = self.context.sigs['self'][self.stmt.func.attr]
+            topics, stored_topics, event, data = pack_logging_topics(event, self.stmt.args, self.context)
+            inargs, inargsize, inarg_start = pack_logging_data(event, data, self.context)
+            return LLLnode.from_list(['seq', inargs, stored_topics, ["log" + str(len(topics)), inarg_start, inargsize] + topics], typ=None, pos=getpos(self.stmt))
```

Comment:
Just for readability could we add some new lines after the return statements ;) I also had another idea, now that we have our own Stmt class we could perhaps replace all those isinstance() calls with self.stmt with `self.is_attribute` and `self.func_name` and `self.is_func` properties etc.. Of course this all to your liking, just a small style thing.

Reply:
Could you write a quick example for how `self.is_attribute` would work?  Would it be syntactic sugar for `isinstance`?

Reply:
So the idea was just as a bunch of handy aliases:

```

class Stmt():

    @property
    def is_attribute(self):
        return isinstance(self.stmt.func, ast.Attribute)

    @property
    def is_name(self):
        return isinstance(self.stmt, ast.Name) 

    @property
    def func_id(self):
        return getattr(self.stmt.func.value, 'id', None)
```